### PR TITLE
Additions to the parsing changes

### PR DESCRIPTION
**Pull request by larshg**
Remove boost dependency

Added a split function.

Added two test to parse X parameters.

### CODE REVIEW

**Comment by larshg** [comment]
I can also add it as a separate PR to main repository?

File: common/src/parse.cpp [hunk 1]
```diff
@@ -45,7 +45,24 @@
...
+{
+  std::vector<std::string> tokens;
+  std::size_t start = text.find_first_not_of(delims), end = 0;
+
+  while ((end = text.find_first_of(delims, start)) != std::string::npos)
+  {
+    tokens.push_back(text.substr(start, end - start));
+    start = text.find_first_not_of(delims, end);
+  }
+  if (start != std::string::npos)
```
Comment (kunaltyagi) [inline common/src/parse.cpp]: Codacy Issue found: [MISRA 15.6 rule](https://app.codacy.com/manual/kunaltyagi/pcl/pullRequest?prid=4957407)

**Comment by kunaltyagi** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 1
           
```


See the complete overview on [Codacy](https://app.codacy.com/manual/kunaltyagi/pcl/pullRequest?prid=4957407&bid=16381951)

**Comment by kunaltyagi** [comment]
The issue is that Boost dependency is not going anywhere, anytime soon. Getting rid of boost is also a bit counter-productive since it is a nice set of tools. If you take a grep search for boost, you'd realize that boost usage is huge in PCL.

The aim is not to remove Boost, but to use `std` where possible instead, as such I don't know how this will be received by others. I myself wouldn't say yes even though it's just 20-30 lines.

But, if you can explain why removing boost is a helpful in this scenario, then this might get accepted

**Pull request closed without merging**